### PR TITLE
Add SlimeTech compatibility

### PR DESCRIPTION
**Pull request by ghost**
## Short Description
Add SlimeTech compatibility
Change register logic

## Additions/Changes/Removals
Added SlimeTech rubber item
Changed register logic

## Related Issues
N/A

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

**Comment by J3fftw1** [comment]
Have you tested this?
im 99% sure this will break recipes in litexpansions that require item id rubber.

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 1]
```diff
@@ -43,6 +43,7 @@ final class ItemSetup {
 
     static final ItemSetup INSTANCE = new ItemSetup();
     private final ItemStack glass = new ItemStack(Material.GLASS);
+    private final ItemStack rubberItem;
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: can't be final

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 2]
```diff
@@ -87,17 +88,19 @@ private void registerMachines() {
...
-    //Disable when SlimyTreeTaps exists
+    // Register only if SlimeTech and SlimyTreeTaps do not exist
     private void registerRubber() {
-        if (!Bukkit.getServer().getPluginManager().isPluginEnabled("SlimyTreeTaps")) {
-            //Rubber
+        if (Bukkit.getServer().getPluginManager().isPluginEnabled("SlimeTech")) {
+            this.rubberItem = SlimefunItem.getById("SLIMETECH_RUBBER").getItem();
+        }else if (Bukkit.getServer().getPluginManager().isPluginEnabled("SlimyTreeTaps")) {
+            this.rubberItem = SlimefunItem.getById("RUBBER").getItem();
+        }else {
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
        } else if (Bukkit.getServer().getPluginManager().isPluginEnabled("SlimyTreeTaps")) {
            this.rubberItem = SlimefunItem.getById("RUBBER").getItem();
        } else {
```

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 3]
```diff
@@ -87,17 +88,19 @@ private void registerMachines() {
         new Converter().register(plugin);
     }
 
-    //Disable when SlimyTreeTaps exists
+    // Register only if SlimeTech and SlimyTreeTaps do not exist
     private void registerRubber() {
-        if (!Bukkit.getServer().getPluginManager().isPluginEnabled("SlimyTreeTaps")) {
-            //Rubber
+        if (Bukkit.getServer().getPluginManager().isPluginEnabled("SlimeTech")) {
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: Why are we prioritising SlimeTech here? SlimyTreeTaps is in 300-400 servers.

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: We shouldnt priortize any the reason this check is here to add support for all rubber if there is any
we should support both imo

**Comment by ghost** [comment]
I'll make the necessary code changes in sometime and test it. Also just a clarification does prioritising meaning putting SlimeTech first in the `if` statement? If yes then I'll switch it to SlimyTreeTaps first.

**Comment by J3fftw1** [comment]
> I'll make the necessary code changes in sometime and test it. Also just a clarification does prioritising meaning putting SlimeTech first in the `if` statement? If yes then I'll switch it to SlimyTreeTaps first.

yes :d

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 4]
```diff
@@ -43,6 +43,7 @@ final class ItemSetup {
 
     static final ItemSetup INSTANCE = new ItemSetup();
     private final ItemStack glass = new ItemStack(Material.GLASS);
+    private ItemStack rubberItem;
     private final SlimefunAddon plugin = LiteXpansion.getInstance();
     private boolean initialised;
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
    private final SlimefunAddon plugin = LiteXpansion.getInstance();
    private boolean initialised;
    private ItemStack rubberItem;
```

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 5]
```diff
@@ -87,17 +88,23 @@ private void registerMachines() {
...
+        } else {
+            rubber = Items.RUBBER;
             registerNonPlaceableItem(Items.RUBBER, RubberSynthesizer.RECIPE_TYPE, SlimefunItems.OIL_BUCKET);
             new RubberSynthesizer().register(plugin);
         }
+        this.rubberItem = rubber;
     }
 
     private void registerMiscItems() {
-        final ItemStack rubberItem = SlimefunItem.getById("RUBBER").getItem();
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:100]: please keep the name rubberItem here and rename the other variable.

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java (hunk 5 above)
Comment (ghost) [reply src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:100]: Will do

**Comment by ghost** [comment]
dun

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 6]
```diff
@@ -43,9 +43,9 @@ final class ItemSetup {
 
     static final ItemSetup INSTANCE = new ItemSetup();
     private final ItemStack glass = new ItemStack(Material.GLASS);
-    private ItemStack rubberItem;
     private final SlimefunAddon plugin = LiteXpansion.getInstance();
     private boolean initialised;
+    private ItemStack rubber;
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
    private ItemStack rubberItem;
```

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 7]
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
 
     // Register only if SlimyTreeTaps and SlimeTech do not exist
     private void registerRubber() {
-        ItemStack rubber;
+        ItemStack rubberItem;
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
```

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 8]
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
...
         if (Bukkit.getPluginManager().isPluginEnabled("SlimyTreeTaps")) {
-            rubber = SlimefunItem.getById("RUBBER").getItem();
+            rubberItem = SlimefunItem.getById("RUBBER").getItem();
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
            this.rubberItem = SlimefunItem.getById("RUBBER").getItem();
```

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 9]
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
...
         } else if (Bukkit.getPluginManager().isPluginEnabled("SlimeTech")) {
-            rubber = SlimefunItem.getById("SLIMETECH_RUBBER").getItem();
+            rubberItem = SlimefunItem.getById("SLIMETECH_RUBBER").getItem();
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
            this.rubberItem = SlimefunItem.getById("SLIMETECH_RUBBER").getItem();
```

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 10]
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
...
         } else {
-            rubber = Items.RUBBER;
+            rubberItem = Items.RUBBER;
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
            this.rubberItem = Items.RUBBER;
```

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 11]
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
...
             registerNonPlaceableItem(Items.RUBBER, RubberSynthesizer.RECIPE_TYPE, SlimefunItems.OIL_BUCKET);
             new RubberSynthesizer().register(plugin);
         }
-        this.rubberItem = rubber;
+        this.rubber = rubberItem;
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
```

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 12]
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
...
     }
 
     private void registerMiscItems() {
-        ItemStack rubber = this.rubberItem;
+        ItemStack rubberItem = this.rubber;
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
```

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 13]
```diff
@@ -146,7 +146,7 @@ private void registerMiscItems() {
         );
 
         registerNonPlaceableItem(Items.TIN_CABLE, RecipeType.ENHANCED_CRAFTING_TABLE,
-            rubber, Items.UNINSULATED_TIN_CABLE
+            rubberItem, Items.UNINSULATED_TIN_CABLE
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
            this.rubberItem, Items.UNINSULATED_TIN_CABLE
```

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 14]
```diff
@@ -158,7 +158,7 @@ private void registerMiscItems() {
         );
 
         registerNonPlaceableItem(Items.COPPER_CABLE, RecipeType.ENHANCED_CRAFTING_TABLE,
-            rubber, Items.UNINSULATED_COPPER_CABLE
+            rubberItem, Items.UNINSULATED_COPPER_CABLE
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
            this.rubberItem, Items.UNINSULATED_COPPER_CABLE
```

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 15]
```diff
@@ -175,7 +175,7 @@ Items.TIN_ITEM_CASING, new ItemStack(Material.REDSTONE), Items.TIN_ITEM_CASING
         );
 
         registerNonPlaceableItem(Items.GOLD_CABLE, RecipeType.ENHANCED_CRAFTING_TABLE,
-            rubber, Items.UNINSULATED_GOLD_CABLE
+            rubberItem, Items.UNINSULATED_GOLD_CABLE
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]:
```suggestion
            this.rubberItem, Items.UNINSULATED_GOLD_CABLE
```

**Comment by ghost** [comment]
Done, I thought if you do something more than once (`this.rubberItem`) it would be better to create a local variable instead. I fixed it.

**Review by J3fftw1 — CHANGES_REQUESTED** [review]
rename the global variable to rubber. The diff still is huge.

**Comment by ghost** [comment]
Walshy just told me to rename to rubber

**Comment by J3fftw1** [comment]
ugh @WalshyDev we have conflicting reviews
there is no need to make the diff bigger then it is

**Comment by ghost** [comment]
Am I supposed to create local variables or nah? Also `rubber` or `rubberItem`

**Comment by WalshyDev** [comment]
The diff literally cannot get smaller..

**Comment by WalshyDev** [comment]
Changing to rubber would be MORE not less.

**Review by J3fftw1 — APPROVED** [review]
LGTM

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**